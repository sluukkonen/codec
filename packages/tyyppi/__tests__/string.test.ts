import { string } from "../src/index.js"
import { expectParseFailure, expectParseSuccess } from "./helpers.js"

test("should parse strings", () => {
  expectParseSuccess(string, "")
  expectParseSuccess(string, "a")
  expectParseSuccess(string, "123")
})

test("should fail to parse non-strings", () => {
  expectParseFailure(string, 0)
})
