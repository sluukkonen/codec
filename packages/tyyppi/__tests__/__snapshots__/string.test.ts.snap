// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should fail to parse non-strings 1`] = `
[
  {
    "code": "invalid_type",
    "expected": "string",
    "message": "Expected a string, received a number",
    "path": [],
    "received": "number",
  },
]
`;
